src/rdb.c

```diff
@@ -1436,6 +1436,48 @@ int rdbSaveRioWithEOFMark(int req, rio *rdb, int *error, rdbSaveInfo *rsi) {
     return C_ERR;
 }
 
+/* Save DB to the file. Similar to rdbSave() but this function won't use a
+ * temporary file and won't update the metrics. */
+int rdbSaveToFile(const char *filename) {
```

Comment:
maybe there's a cleaner way to avoid code duplication by changing rdbSave to use rdbSaveToFile?
i.e. it generates the temp file name, and later does the rename and update metrics..

Reply:
I'm having a bit hard time to reuse this code. Currently, `rdbSave()` updates the metrics and calls `stopSaving()`. If we call `rdbSaveToFile()` first, it will call `stopSaving()` before updating the metrics. I thought this might be a breaking change (module callback will be called before updating the metrics), that's why I went with separate functions. If you think this is not important, I can make the change.

Reply:
you mean `dirty`, `lastsave`, and `lastbgsave_status`?
i don't see a problem with the last two, modules do get an indication of success in the callback.
maybe the only risk here is that is set reset the dirty counter after the callback, then it'll undo any `dirty++` that's done inside the callback, i.e. if the module would do call RM_Call from that callback.
bottom line, i think modules are not suppose to do any such things from this callback, so re-ordering this seems ok to me.

Reply:
Yeah, I was worried about things like if module wants to read `lastsave` timestamp in the callback (via `RM_Call("info")` maybe?). If we don't care about these, then I'll update the PR.

Reply:
@oranagra added a commit for this. Please take a look.

Reply:
so i see you moved startSaving and stopSaving to the outer function, so that concern is resolved (no implications on the existing behavior, right?)
i would be ok with putting them inside the inner function too, but no objections to what you've done

Reply:
Yes, I moved them to the outer function. Maybe a tiny implication, previously, we were calling `startSaving()` after `fopen()`. Now, `startSaving()` will be called after `fopen()`. 

If we put `startSaving() / stopSaving()` into the internal function, for `rdbSave()`, we would call `stopSaving(1)` (with success code) before doing rename + fsync dir. So, right now, it will only call `stopSaving()` after rename + fsync.

Reply:
ok. LGTM.